python/ray/serve/tests/test_standalone2.py

```diff
@@ -561,6 +563,68 @@ def test_controller_recover_and_deploy(self, client: ServeControllerClient):
         # Ensure config checkpoint has been deleted
         assert client.get_serve_status().app_status.deployment_timestamp == 0
 
+    @pytest.mark.parametrize(
+        "option_to_update,config_update",
+        [
+            ("num_replicas", True),
+            ("autoscaling_config", True),
+            ("user_config", True),
+            ("ray_actor_options", False),
+        ],
+    )
+    def test_deploy_config_update(
+        self, client: ServeControllerClient, option_to_update: str, config_update: bool
+    ):
+        """
+        Check that replicas stay alive when lightweight config updates are made and
+        replicas are torn down when code updates are made.
+        """
+
+        def deployment_running():
+            serve_status = client.get_serve_status()
+            return (
```

Comment:
Can we also track `deployment_timestamps` via this function? E.g. we could pass in the last timestamp as an argument and check that the current `deployment_timestamp` is greater. That would ensure that the deployment is running and has been updated.

Reply:
Note that when we haven't deployed a config yet, the `deployment_timestamp` is 0.

Reply:
I synced with @zcin offline, and it turns out that after `deploy_app` runs, the `deployment_timestamp` should be updated anyways, so this isn't an issue.